module/zfs/zfs_chksum.c

```diff
@@ -155,6 +155,11 @@ chksum_run(chksum_stat_t *cs, abd_t *abd, void *ctx, int round,
 		size = 1<<22; loops = 1; break;
 	}
 
+	/* skip benchmarking for some architectures */
+	if (size > SPA_MAXBLOCKSIZE) {
```

Comment:
Ups, seems that `SPA_MAXBLOCKSIZE` does not - want I thought it is for :-(
But `zfs_max_recordsize` is only 1MB for x86 32bit? So `arm` and `mips32` would not be catched...
I have to search for a better constant... the value of `4m` was choosen by me because of that constant: `SPL_KMEM_CACHE_MAX_SIZE` within `include/os/linux/spl/sys/kmem_cache.h` ... because the call can not fail ... I can not check for some NULL return of kmem_cache_alloc() ... what next?

Or should I really just downgrade the `4m` to `2m` ... what arch maybe needs `1m` to be the minimum?

Reply:
You are right .... yes.

Reply:
Can the `_ILP32` define be used for selecting all 32bit archs ... and will all other ones have no trouble with 4m ... then it would be a simple ifdef which may be okay for that.
But of cause... do we need the benchmark for 4m ?